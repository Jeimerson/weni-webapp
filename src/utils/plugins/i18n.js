import Vue from 'vue';
import VueI18n from 'vue-i18n';

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';

Vue.use(VueI18n);

const languages = {
  en: en,
  'pt-br': pt_br,
  'pt-BR': pt_br,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en',
  fallbackLocale: 'en',
  messages,
});

export default i18n;
