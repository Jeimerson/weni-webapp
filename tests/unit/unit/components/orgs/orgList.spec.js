import { shallowMount, createLocalVue, RouterLinkStub } from '@vue/test-utils';
import Vuex from 'vuex';
import Router from 'vue-router';
import i18n from '@/utils/plugins/i18n';
import OrgList from '@/components/orgs/orgList.vue';
import { org } from '../../../__mocks__';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Router);

const router = new Router();

jest.mock('@/api/request.js', () => {});

describe('orgList.vue', () => {
  let wrapper;
  let store;
  let actions;

  beforeEach(() => {
    actions = {
      getOrgs: jest.fn(),
      deleteOrg: jest.fn(),
      setCurrentOrg: jest.fn(),
      clearCurrentOrg: jest.fn(),
      clearCurrentProject: jest.fn(),
      openModal: jest.fn(),
    };

    store = new Vuex.Store({
      actions,
    });

    wrapper = shallowMount(OrgList, {
      localVue,
      i18n,
      store,
      router,
      mocks: {
        $t: () => 'some specific text',
      },
      stubs: {
        RouterLink: RouterLinkStub,
        OrgListItem: true,
        NewInfiniteLoading: true,
        UnnnicSkeletonLoading: true,
        RightSideBar: true,
      },
    });
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('onNavigateToBilling', () => {
    const spySelectOrg = jest.spyOn(wrapper.vm, 'selectOrg');
    const spyRouter = jest.spyOn(wrapper.vm.$router, 'push');

    wrapper.vm.onNavigateToBilling({ uuid: 12 });

    expect(spySelectOrg).toHaveBeenCalledTimes(1);
    expect(spyRouter).toHaveBeenCalledTimes(1);
  });
});
