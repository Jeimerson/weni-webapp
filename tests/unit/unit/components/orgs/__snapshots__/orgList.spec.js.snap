// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orgList.vue should be rendered properly 1`] = `undefined`;
